hummingbot/core/data_type/user_stream_tracker_data_source.py

```diff
@@ -44,7 +44,8 @@ async def listen_for_user_stream(self, output: asyncio.Queue):
             try:
                 self._ws_assistant = await self._connected_websocket_assistant()
                 await self._subscribe_channels(websocket_assistant=self._ws_assistant)
-                await self._ws_assistant.ping()  # to update last_recv_timestamp
+                if isinstance(self._ws_assistant, WSAssistant):  # adding check for connectors with their own WS client
+                    await self._ws_assistant.ping()  # to update last_recv_timestamp
```

Comment:
Instead of adding an `if` statement we should define a new method to send the ping, that by default will do `await ws_assistant.ping()` (ws_assistant will be received as a parameter). The method can be overwritten in subclasses to do something different

Reply:
Added a separate function to ping in the superclass, which can be overriden.